src/addr6.rs

```diff
@@ -200,24 +197,34 @@ impl AsMut<[u8]> for MacAddr6 {
     }
 }
 
-#[cfg(feature = "std")]
-mod std {
-    use std::fmt;
-
-    use super::MacAddr6;
-
-    impl fmt::Display for MacAddr6 {
-        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-            f.write_fmt(format_args!(
-                    // Canonical form
-                    "{:02X}-{:02X}-{:02X}-{:02X}-{:02X}-{:02X}",
-                    self.0[0],
-                    self.0[1],
-                    self.0[2],
-                    self.0[3],
-                    self.0[4],
-                    self.0[5],
-                ))
+impl fmt::Display for MacAddr6 {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        for (i, n) in self.0.iter().enumerate() {
```

Comment:
I feel like while this implementation is more flexible, it will generate much more unefficient output compared to the previous one; check out this example: https://godbolt.org/z/CJziTM

Previous implementation (called `MacAddr6Old`) generates straightforward asm code, while current one (`MacAddr6New`) is full of conditional jumps and multiple `call` instructions.
How about we hardcode the placeholders directly into the code? Smth like this:

```rust
if f.sign_minus() {
    f.write_fmt(format_args!("{:02X}-{:02X}-{:02X}-{:02X}-{:02X}-{:02X}"))
} else if f.alternate() {
    f.write_fmt(format_args!("{:02X}:{:02X}:{:02X}:{:02X}:{:02X}:{:02X}"))
}
```
```